src/utils/promiseUtils.ts

```diff
@@ -24,3 +25,12 @@ export async function delay(ms: number, token?: vscode.CancellationToken): Promi
         }
     });
 }
+
+export async function getCancelPromise(token: vscode.CancellationToken, message?: string): Promise<never> {
```

Comment:
What do you think about subclassing `CancellationTokenSource` and adding this method to the new class? This way we can deal with just one thing in the rest of the code

Reply:
That's an interesting idea, I'll think about it.

Reply:
@philliphoff pointed out that in some cases (for example, `DebugConfigurationProvider.resolveDebugConfiguration`), we do not have `CancellationTokenSource`, only `CancellationToken`. So it seemed best to have both this method, and a derivative class.